content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md

```diff
@@ -693,7 +675,83 @@ data:
 
 - You can't use ConfigMaps for {{< glossary_tooltip text="static pods" term_id="static-pod" >}}, because the Kubelet does not support this.
 
+### Optional ConfigMaps
+
+A ConfigMap reference may be marked "optional".
+If the ConfigMap is non-existent, the mounted volume will be empty.
+If the ConfigMap exists, but the referenced key is non-existent the path will be absent beneath the mount point.
 
+#### Optional ConfigMap in environment variables
+
+There might be situations where environment variables are not always required.
+These environment variables can be marked as optional in a pod like so:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dapi-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: gcr.io/google_containers/busybox
+      command: [ "/bin/sh", "-c", "env" ]
+      env:
+        - name: SPECIAL_LEVEL_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: a-config
+              key: akey
+              optional: true
+  restartPolicy: Never
+```
+
+When this Pod is run, the output will be empty.
+
+#### Optional ConfigMap via volume plugin
+
+Volumes and files provided by a ConfigMap can be also be marked as optional.
+The ConfigMap or the key specified does not have to exist.
+The mount path for such items will always be created.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dapi-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: gcr.io/google_containers/busybox
+      command: [ "/bin/sh", "-c", "ls /etc/config" ]
+      volumeMounts:
+      - name: config-volume
+        mountPath: /etc/config
+  volumes:
+    - name: config-volume
+      configMap:
+        name: no-config
+        optional: true
+  restartPolicy: Never
+```
+
+When this pod is run, the output will be:
+
+```shell
```

Comment:
(nit) the output is not shell

Reply:
Updated in https://github.com/kubernetes/website/pull/35160/commits/7e16543b9dbe6771017730199b62e2b99c18803e